### PR TITLE
Add table-chunker

### PR DESCRIPTION
**Pull request by dhruvkaliraman7**
1. Add a "column_headers" field to the table element, to store the column headers.
2. While Splitting elements for tables, split at a new line character, and attach column headers and other metadata to table chunk.

### CODE REVIEW

File: lib/sycamore/sycamore/data/table.py [hunk 1]
```diff
@@ -305,6 +308,7 @@ def to_pandas(self) -> DataFrame:
             index=None,
             columns=flattened_header if max_header_prefix_row >= 0 else None,
         )
+        self.column_headers = flattened_header if max_header_prefix_row >= 0 else []
```
Comment (HenryL27) [inline lib/sycamore/sycamore/data/table.py:311]: I really don't like mutating the table object in this method. Can we avoid this or do it in like init or something?

File: lib/sycamore/sycamore/data/table.py [hunk 2]
```diff
@@ -112,6 +114,31 @@ def __init__(self, cells: list[TableCell], caption: Optional[str] = None):
...
+                    if cell.is_header and cell.rows[0] <= max_header_prefix_row:
+                        for col in cell.cols:
+                            header_array[cell.rows[0], col] = cell.content
+                        for row in cell.rows[1:]:
+                            for col in cell.cols:
+                                header_array[row, col] = ""
+
+            self.column_headers = []
+            for npcol in header_array.transpose():
+                self.column_headers.append(" | ".join(OrderedDict.fromkeys((c for c in npcol if c not in [None, ""]))))
```
Comment (HenryL27) [inline lib/sycamore/sycamore/data/table.py:140]: any chance we can factor this as a function in common with the pandafication function too?

**Review by HenryL27 — COMMENTED** [review]
now we've ended up with a bunch of duplicated code...

File: lib/sycamore/sycamore/data/table.py (hunk 2 above)
Comment (dhruvkaliraman7) [reply lib/sycamore/sycamore/data/table.py]: done

File: lib/sycamore/sycamore/data/table.py [hunk 3]
```diff
@@ -275,7 +255,7 @@ def from_html(cls, html_str: Optional[str] = None, html_tag: Optional[Tag] = Non
     # we speculate that duplication may create confusion, so we default to only displaying a cells
     # content for the first row/column for which it is applicable. The exception is for header rows,
     # where we duplicate values to each columnn to ensure that every column has a fully qualified header.
-    def to_pandas(self) -> DataFrame:
+    def to_pandas(self, column_header: Optional[bool] = False) -> Union[DataFrame, List[str]]:
```
Comment (HenryL27) [inline lib/sycamore/sycamore/data/table.py:258]: this type can just be bool with a default. I don't think you need 3 states
also (obnoxious nit) might call the param `column_header_only`

**Review by HenryL27 — APPROVED** [review]
lgtm, one nit

**Pull request merged**